lib/practice_exercises.rb

```diff
@@ -1,13 +1,42 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n); time that the process takes will depend on the size of the string;
+# Constants are dropped.
+# Space Complexity: O(1), because no new structures are created.
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  i = 0
+  string_array = string.downcase.split("")
+
+  string_array.each do |char|
+    if ["?", " ", ".", ",", "!", ":", ";"].include?(char)
+      string_array.delete(char)
+    end
+  end
+
+  while i < string_array.length
+    if string_array[i] != string_array[-(i + 1)]
+      return false
+    end
+    i += 1
+  end
+
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2), because there are two nested loops, both of which have 
+# a complexity of O(n) that depends on size of the input.
+# Space Complexity: O(1), because no new structures are created.
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word.  Otherwise 👍